File: frontend/src/services/roleService.ts
import { apiClient } from '@/config/apiClient';
import type { ApiResponse, PageResponse, RoleRequest, RoleResponse, TableParams } from '@/types';

export const roleService = {
  getAll: (params?: TableParams): Promise<ApiResponse<PageResponse<RoleResponse>>> => {
    return apiClient.get('/roles', { params });
  },

  getAllActive: (): Promise<ApiResponse<RoleResponse[]>> => {
    return apiClient.get('/roles/active');
  },

  getById: (id: number): Promise<ApiResponse<RoleResponse>> => {
    return apiClient.get(`/roles/${id}`);
  },

  create: (data: RoleRequest): Promise<ApiResponse<RoleResponse>> => {
    return apiClient.post('/roles', data);
  },

  update: (id: number, data: RoleRequest): Promise<ApiResponse<RoleResponse>> => {
    return apiClient.put(`/roles/${id}`, data);
  },

  delete: (id: number): Promise<ApiResponse<void>> => {
    return apiClient.delete(`/roles/${id}`);
  },
};

File: backend/src/main/java/com/scim/service/RoleService.java
package com.scim.service;

import com.scim.dto.role.request.RoleRequest;
import com.scim.dto.response.PageResponse;
import com.scim.dto.role.response.RoleResponse;
import org.springframework.data.domain.Pageable;

import java.util.List;

public interface RoleService {
    PageResponse<RoleResponse> getAll(String search, Pageable pageable);
    List<RoleResponse> getAllActive();
    RoleResponse getById(Long id);
    RoleResponse create(RoleRequest request);
    RoleResponse update(Long id, RoleRequest request);
    void delete(Long id);
}
